src/Raven.Client/Connection/Operation.cs

```diff
@@ -25,15 +29,24 @@ public Operation(AsyncServerClient asyncServerClient, long id)
         {
             _asyncServerClient = asyncServerClient;
             _id = id;
-            Task.Run(Initialize);
+
+            Task.Factory.StartNew(Initialize);
         }
 
         private async Task Initialize()
         {
-            await _asyncServerClient.changes.Value.ConnectionTask.ConfigureAwait(false);
-            var observableWithTask = _asyncServerClient.changes.Value.ForOperationId(_id);
-            _subscription = observableWithTask.Subscribe(this);
-            await FetchOperationStatus();
+            try
+            {
+                await _asyncServerClient.changes.Value.ConnectionTask.ConfigureAwait(false);
+                var observableWithTask = _asyncServerClient.changes.Value.ForOperationId(_id);
+                _subscription = observableWithTask.Subscribe(this);
+                await FetchOperationStatus().ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                _result.TrySetException(e);
+                throw;
```

Comment:
The problem is that then this raised the UnhandledTaskException, and we get bug reports on that.